liquibase-standard/src/main/java/liquibase/command/core/DbDocCommandStep.java

```diff
@@ -1,64 +1,142 @@
-
     public static final CommandArgumentDefinition<String> SCHEMAS_ARG;
+    public static final CommandArgumentDefinition<String> LABEL_FILTER_ARG;
+    public static final CommandArgumentDefinition<String> CONTEXTS_ARG;
+    public static final CommandArgumentDefinition<CatalogAndSchema[]> CATALOG_AND_SCHEMAS_ARG;
 
     static {
         CommandBuilder builder = new CommandBuilder(COMMAND_NAME);
-        CHANGELOG_FILE_ARG = builder.argument(CommonArgumentNames.CHANGELOG_FILE, String.class)
-                .description("The root changelog").required().build();
-        URL_ARG = builder.argument(CommonArgumentNames.URL, String.class).required()
-                .description("The JDBC database connection URL").build();
-        DEFAULT_SCHEMA_NAME_ARG = builder.argument("defaultSchemaName", String.class)
-                .description("The default schema name to use for the database connection").build();
-        DEFAULT_CATALOG_NAME_ARG = builder.argument("defaultCatalogName", String.class)
-                .description("The default catalog name to use for the database connection").build();
-        DRIVER_ARG = builder.argument("driver", String.class)
-                .description("The JDBC driver class").build();
-        DRIVER_PROPERTIES_FILE_ARG = builder.argument("driverPropertiesFile", String.class)
-                .description("The JDBC driver properties file").build();
-        USERNAME_ARG = builder.argument(CommonArgumentNames.USERNAME, String.class)
-                .description("The database username").build();
-        PASSWORD_ARG = builder.argument(CommonArgumentNames.PASSWORD, String.class)
-                .description("The database password")
-                .setValueObfuscator(ConfigurationValueObfuscator.STANDARD)
+        OUTPUT_DIRECTORY_ARG = builder.argument("outputDirectory", String.class)
+                .required()
+                .description("The directory where the documentation is generated")
                 .build();
-        OUTPUT_DIRECTORY_ARG = builder.argument("outputDirectory", String.class).required()
-                .description("The directory where the documentation is generated").build();
-
         SCHEMAS_ARG = builder.argument("schemas", String.class)
-                .description("Database schemas to include objects from in reporting").build();
+                .description("Database schemas to include objects from in reporting")
+                .build();
+        CATALOG_AND_SCHEMAS_ARG = builder.argument("catalogAndSchemas", CatalogAndSchema[].class)
+                .hidden()
+                .optional()
+                .build();
+        LABEL_FILTER_ARG = builder.argument("labelFilter", String.class)
+                .addAlias("labels")
+                .description("Changeset labels to match")
+                .build();
+        CONTEXTS_ARG = builder.argument("contexts", String.class)
+                .description("Changeset contexts to match")
+                .build();
```

Comment:
Can you remove these and get these from the DatabaseChangelogCommandStep dependency? Like this:

```
        ChangeLogParameters changeLogParameters = (ChangeLogParameters) commandScope.getDependency(ChangeLogParameters.class);
        Contexts contexts = changeLogParameters.getContexts();
        LabelExpression labels = changeLogParameters.getLabels();
```

Reply:
Great catch, really cleans things up!